### PR TITLE
Feature/566 support agents can create new invitations

### PR DESCRIPTION
**Pull request by despo**
## Trello card URL:
https://trello.com/c/fZaY18hu/566-23-support-agents-can-create-new-invitations-for-hiring-staff

## Changes in this PR:
- enables support agents to invite users by first name, family name and email address
- triggers Notify, DSI and Authorisation service requests
- informs support user of errors and to seek help from dev team

### CODE REVIEW

**Comment by despo** [comment]
[2.3 Support agents can create new invitations for hiring staff](https://trello.com/c/fZaY18hu/566-23-support-agents-can-create-new-invitations-for-hiring-staff)

**Comment by lozette** [comment]
Thank you for your help on this! Looks good to me but @tahb should do the final review as we've both worked on it

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      @user.save
+
+      begin
+        InviteService.new(user: @user.to_invite_params, school: @school.to_invite_params).call
+        flash[:notice] = "#{@user.full_name} successfully invited to #{@school.name}"
+      rescue RuntimeError => e
+        ref = SecureRandom.hex.first(10)
+        Logger.new($stdout).error("(#{ref}) #{e.message}")
+
+        flash[:danger] = "There was a problem inviting #{@user.full_name} to #{@school.name}.
```
Comment (tahb) [inline app/controllers/users_controller.rb]: To get this from the logs more than 2 weeks in the future of the event, we'll need the timestamp together with this reference to find it in our Papertrail archive. Suggest we add it here too as we're asking the user to quote a reference to us via a ticket.

Comment (despo) [reply app/controllers/users_controller.rb]: @tahb  I can do this. However, I have removed this code as part of my current task. 
By logging the error to the audit both the support agents and the devs can see what went wrong as well as the exact time, which should enable tracing the entry in the logs if more debugging is needed.

error_message_and_tracking
audit_view

Comment (tahb) [reply app/controllers/users_controller.rb]: Understood!

File: .env.example [hunk 2]
```diff
@@ -1,7 +1,10 @@
-# Add account credentials and API keys here.
-# This file should be listed in .gitignore to keep your settings secret!
-# Each entry sets a local environment variable.
-# For example, setting:
-# GMAIL_USERNAME=Your_Gmail_Username
-# makes 'Your_Gmail_Username' available as ENV["GMAIL_USERNAME"]
-
+TVA_URL=https://tva-staging.herokuapp.com
```
[the remark this answers is not in the document]
Comment (despo) [reply .env.example]: 🤔 My environment seems to be utilising the values from https://github.com/dxw/teacher-vacancy-support/blob/develop/config/secrets.yml#L23

[the remark this answers is not in the document]
Comment (despo) [reply .env.example]: @tahb could your local  be using production instead of development for some reason?

[the remark this answers is not in the document]
Comment (tahb) [reply .env.example]: Yes so is mine, I had values left over in my .env and I jumped to a conclusion.

**Comment by despo** [comment]
@tahb I have tackled this as part of the next PR, specifically https://github.com/dxw/teacher-vacancy-support/pull/11/commits/ad2e2fe092b15adfc6677a938c7cb9a1f0a427a9

To enable this to work without issues I had to make a tweak, which is retrieving a user by email before creating the entry. (I did spend sometime trying to break this down to more steps but it was adding a lot of complexity). 
I also enforced email address uniqueness validation.

**Pull request merged**